ar_module2_8.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+var myDate = (function iffe() {
+  /* eslint-disable no-console */
+  /* eslint-disable no-unused-vars */
+  var privateDate = new Date(); // make this available to both get and set
+  var dd = privateDate.getDate();
+  var mm = privateDate.getMonth() + 1; //January is 0!
+  var yyyy = privateDate.getFullYear();
+
+  if (dd < 10) {
+    dd = '0' + dd;
+  }
+  if (mm < 10) {
+    mm = '0' + mm;
+  }
+
+  var checkDate = mm + dd + yyyy;
+  var formatDate = mm + '/' + dd + '/' + yyyy;
+
+  function setDate(newDate) {  // make a function for milli
+
+    if (newDate == undefined) {
+      var defaultDate = privateDate;
+      console.log(defaultDate);
+      // Done - myDate.setDate() if no argument given
+    }
+
+    else {  // only using newMilli in use
+      var milliDate = new Date();
```

Comment:
Refer back to the requirements for the setDate method. If the user passes in a millisecond value or a date instance, you want to set your internal date variable to a new instance of a date object that is instantiated with either of those values. For example, if `dateObj.setDate(123456789);` is called, you want to set your internal date to a new date that uses that value, but you should check that value first:

`if (typeof newDate === 'number') { privateDate = new Date(newDate); }`

Likewise, if newDate is an instance of a Date object, do the same:

`if (newDate instanceof Date) { privateDate = new Date(newDate); }`